tests/integration/suite/actors/reminders/serialization/protobuf.go

```diff
@@ -0,0 +1,115 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliei.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serialization
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/process/sqlite"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(serializationProtobuf))
+}
+
+// serializationProtobuf tests:
+// - The ability for daprd to read reminders serialized as JSON and protobuf
+// - That reminders are serialized to protobuf when the Actors API level in the cluster is >= 20
+type serializationProtobuf struct {
+	daprd   *daprd.Daprd
+	srv     *prochttp.HTTP
+	handler *httpServer
+	place   *placement.Placement
+	db      *sqlite.SQLite
+}
+
+func (i *serializationProtobuf) Setup(t *testing.T) []framework.Option {
+	// Init placement with minimum API level of 20
+	i.place = placement.New(t, placement.WithMinAPILevel(20))
+
+	// Create a SQLite database and ensure state tables exist
+	i.db = sqlite.New(t, sqlite.WithActorStateStore(true))
+	i.db.CreateStateTables(t)
+
+	// Init daprd and the HTTP server
+	i.handler = &httpServer{}
+	i.srv = prochttp.New(t, prochttp.WithHandler(i.handler.NewHandler()))
+	i.daprd = daprd.New(t,
+		daprd.WithResourceFiles(i.db.GetComponent()),
+		daprd.WithPlacementAddresses("localhost:"+strconv.Itoa(i.place.Port())),
+		daprd.WithAppPort(i.srv.Port()),
+		// Daprd is super noisy in debug mode when connecting to placement.
+		daprd.WithLogLevel("info"),
+	)
+
+	// Store a reminder encoded as JSON before starting the test
+	// This will assert the ability to "upgrade" from JSON to Protobuf
+	queryCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	now := time.Now().UTC().Format(time.RFC3339)
+	_, err := i.db.GetConnection(t).ExecContext(queryCtx, fmt.Sprintf(`
+INSERT INTO state VALUES
+  ('actors||myactortype','[{"registeredTime":"%[1]s","period":"2m","actorID":"myactorid","actorType":"myactortype","name":"oldreminder","dueTime":"0"}]',0,'e467f810-4e93-45ed-85d9-e68d9fc7af4a',NULL,'%[1]s'),
+  ('actors||myactortype||metadata','{"id":"00000000-0000-0000-0000-000000000000","actorRemindersMetadata":{"partitionCount":0}}',0,'e82c5496-ae32-40a6-9578-6a7bd84ff331',NULL,'%[1]s');
+`, now))
+	require.NoError(t, err)
```

Comment:
This doesn't belong here- the code smell is the `context.Background()`.

This should be moved to a `WithRunExec` option which is executed on the process `Run`. This will then also have the proper test context a available.